java/src/jmri/jmrit/display/layoutEditor/LayoutEditorChecks.java

```diff
@@ -0,0 +1,199 @@
+                if (!flag) {
+                    checkUnBlockedTracksMenu.addSeparator();
+                    flag = true;
+                }
+                // add this layout to the check un-connected menu
+                JMenuItem jmi = new JMenuItem(layoutTrack.getName());
+                checkUnBlockedTracksMenu.add(jmi);
+                jmi.addActionListener((ActionEvent event) -> {
+                    JMenuItem menuItem = (JMenuItem) event.getSource();
+                    String menuItemName = menuItem.getText();
+                    doCheckUnBlockedTracksMenuItem(menuItemName);
+                });
+            }
+        }
+    }
+
+    // action to be performed when checkUnConnectedTracksMenuItem is clicked
+    private void doCheckUnConnectedTracksMenuItem(String menuItemName) {
+        log.info("docheckUnConnectedTracksMenuItem({})", menuItemName);
+        layoutEditor.clearSelectionGroups();
+        LayoutTrack layoutTrack = layoutEditor.getFinder().findObjectByName(menuItemName);
+        if (layoutTrack != null) {
+            layoutEditor.amendSelectionGroup(layoutTrack);
+        }
+    }
+
+    // action to be performed when checkUnBlockedTracksMenuItem is clicked
+    private void doCheckUnBlockedTracksMenuItem(String menuItemName) {
+        log.info("doCheckUnBlockedTracksMenuItem({})", menuItemName);
+
+        layoutEditor.clearSelectionGroups();
+        LayoutTrack layoutTrack = layoutEditor.getFinder().findObjectByName(menuItemName);
+        if (layoutTrack != null) {
+            layoutEditor.setSelectionRect(layoutTrack.getBounds());
+            layoutEditor.clearSelectionGroups();
+            layoutEditor.amendSelectionGroup(layoutTrack);
+
+            Point2D p = layoutTrack.getCoordsCenter();
+            p = MathUtil.multiply(p, layoutEditor.getZoom());
+
+            JPopupMenu popupMenu = layoutTrack.showPopup(new MouseEvent(
+                    layoutEditor.getTargetPanel(), // source
+                    MouseEvent.MOUSE_CLICKED, // id
+                    System.currentTimeMillis(), // when
+                    0, // modifiers
+                    (int) p.getX(), (int) p.getY(), // where
+                    0, // click count
+                    true));                         // popup trigger
+
+            // if we can find the "Edit" menu item go ahead and click it
+            String editMenuItemText = Bundle.getMessage("ButtonEdit");
+            for (int i = 0; i < popupMenu.getComponentCount(); i++) {
+                Component c = popupMenu.getComponent(i);
+                if (c != null) {
+                    if (c instanceof JMenuItem) {
+                        JMenuItem jmi = (JMenuItem) c;
+                        if (jmi.getText().equals(editMenuItemText)) {
+                            // hide the popup menu
+                            popupMenu.setVisible(false);
+                            // and click the "Edit" menu
+                            jmi.doClick();
```

Comment:
I am a little confused here.

Why are we searching for a button and then clicking on it?  Isn't it more efficient to just perform the action of the button?

Reply:
Yes, done!